code/edge/src/main/java/com/adobe/marketing/mobile/RequestBuilder.java

```diff
@@ -156,12 +160,16 @@ JSONObject getConsentPayload(final Event event) {
 		query.setConsentOptions(consentQueryOptions);
 		consents.setQueryOptions(query);
 
-		try {
-			if (xdmPayloads != null && xdmPayloads.get(EdgeConstants.EventDataKey.IDENTITY_MAP) != null) {
-				consents.setIdentityMap((Map<String, Object>) xdmPayloads.get(EdgeConstants.EventDataKey.IDENTITY_MAP));
-			}
-		} catch (ClassCastException e) {
-			Log.debug(LOG_TAG, LOG_SOURCE, "Failed to read IdentityMap from request payload, not a map");
+		final Map<String, Object> identityMap = DataReader.optTypedMap(
+			Object.class,
+			xdmPayloads,
+			EdgeConstants.EventDataKey.IDENTITY_MAP,
+			null
+		);
+		if (Utils.isNullOrEmpty(xdmPayloads)) {
```

Comment:
Should this read `if (Utils.isNullOrEmpty(identityMap)) ...`?

Reply:
good 👀